src/Application/Messenger/Handler/StripePayoutHandler.php

```diff
@@ -113,7 +113,13 @@ public function __invoke(StripePayout $payout): void
             return;
         }
 
-        foreach ($this->getOriginalDonationChargeIds($paidChargeIds, $connectAccountId, $payoutCreated) as $chargeId) {
+        $chargeIds = $this->getOriginalDonationChargeIds($paidChargeIds, $connectAccountId, $payoutCreated);
+
+        if (count($chargeIds) > 0) {
```

Comment:
Might be a bit clearer with the if reversed to make a guard condition like we have above, instead of wrapping the beginTransaction:

```php
        if ($chargeIds === []) {
            $this->logger->info(sprintf(
                'Payout: Exited with no original donation charge IDs for Payout ID %s',
                $payoutId,
            ));
            
            return;
        }
```